randovania/cli/database.py

```diff
@@ -397,66 +400,77 @@ def _find_uncommented_tricks(requirement: Requirement):
+                    for req in trick_resources
+                )
+            ),
+            requirement.comment is not None,
         )
 
 
-def uncommented_trick_usages_logic(args):
+def _flat_trick_usage(requirement: Requirement) -> dict[str, bool]:
+    doc: dict[str, bool] = {}
+    for usage, documented in sorted(set(_find_tricks_usage_documentation(requirement))):
+        doc[usage] = documented and doc.get(usage, True)
+    return doc
+
+
+def _get_area_connection_docs(area: Area) -> dict[str, dict[str, dict[str, bool]]]:
+    paths: dict[str, dict[str, dict[str, bool]]] = {}
+    for source, connections in area.connections.items():
+        paths[source.name] = {}
+        for target, requirement in connections.items():
+            trick_documentation = _flat_trick_usage(requirement)
+            if trick_documentation:
+                paths[source.name][target.name] = trick_documentation
+    return paths
+
+
+def trick_usage_documentation_logic(args: argparse.Namespace) -> None:
     gd = load_game_description(args)
     output_path: Path = args.output_path
 
     lines = []
     for region in gd.region_list.regions:
         lines.append(f"# {region.name}")
         for area in region.areas:
-            paths = {}
-            for source, connections in area.connections.items():
-                paths[source.name] = {}
-                for target, requirement in connections.items():
-                    undocumented = sorted(set(_find_uncommented_tricks(requirement)))
-                    if undocumented:
-                        paths[source.name][target.name] = undocumented
+            paths = _get_area_connection_docs(area)
 
             if paths and any(paths.values()):
-                lines.append(f"## {area.name}")
+                lines.append(f"\n## {area.name}")
                 for source_name, connections in paths.items():
                     if connections:
-                        lines.append(f"### {source_name}")
-                        for target_name, undocumented in connections.items():
-                            if len(undocumented) == 1:
-                                lines.append(f"- [ ] {target_name}: {undocumented[0]}\n")
-                            else:
-                                lines.append(f"#### {target_name}")
-                                for it in undocumented:
-                                    lines.append(f"- [ ] {it}\n")
+                        for target_name, trick_documentation in connections.items():
+                            lines.append(f"### {source_name} -> {target_name}:")
+                            for it, used in trick_documentation.items():
+                                lines.append(f"- [{'X' if used else ' '}] {it}")
 
     output_path.write_text("\n".join(lines))
 
 
-def uncommented_trick_usages_command(sub_parsers):
+def trick_usage_documentation_command(sub_parsers: _SubParsersAction) -> None:
```

Comment:
Do we not have a test for this subcommand? Codecv doesn't seem to report anything at all.

Reply:
probably not, a lot of the CLI things does not have tests

Reply:
would be nice to have tho.

Reply:
Get my very low effort test, all I can have for you today